File: recommendation-analysis/src/main/scala/com/jangz/recommendation/util/StringUtil.scala
package com.jangz.recommendation.util

object StringUtil {
  def limitString(str: String, len: Int, encoding: String): String = {
    var s = str
    var yy: Array[Byte] = s.getBytes(encoding)
    while (s != null && yy.length > len) {
      var sublen: Int = getEncodingLen(yy.length, s.length(), len)
      if (sublen < 0) {
        sublen = s.length() / 2
      }
      s = s.substring(0, sublen)
      yy = s.getBytes(encoding)
    }
    s
  }

  def getEncodingLen(bytelen: Int, strlen: Int, maxlen: Int): Int = {
    strlen - (bytelen - maxlen)
  }

  def clearTitleAll(title: String): String = {
    var result: String = clearTitle(title, "-")
    result = clearTitle(result, "_")
    result
  }

  def clearTitle(title: String, symbol: String): String = {
    if (title == null) {
      return ""
    }
    val index: Int = title.lastIndexOf(symbol)
    if (index < 5) {
      title
    } else {
      clearTitle(title.substring(0, index), symbol)
    }
  }
}

File: recommendation-web/src/main/java/com/jangz/recommendation/util/StringUtil.java
package com.jangz.recommendation.util;

public class StringUtil {

	public static String toFirstUpperCase(String str) {
		if(str == null || str.length() < 1) {
			return "";
		}
		String start = str.substring(0,1).toUpperCase();
		String end = str.substring(1, str.length());
		return start + end;
	}

	public static String limitTitle(String title,int count) {
		if(title != null && title.length() > count) {
			return title.substring(0, count) + "...";
		}
		return title;
	}

	public static String clearTitleAll(String title) {
		String result = clearTitle(title, "-");
		result = clearTitle(result, "_");
		return result;
	}

	public static String clearTitle(String title, String symbol) {
		if(title == null) {
			return "";
		}
		int index = title.lastIndexOf(symbol);
		return index < 5 ? title : clearTitle(title.substring(0, index), symbol);
	}
	
	public static void main(String [] args) {
		String title = "习近平访问欧盟";
		System.out.println(StringUtil.limitTitle(title, 5));
	}
}
